napari/utils/misc.py

```diff
@@ -260,3 +260,53 @@ def all_subclasses(cls: Type) -> set:
     return set(cls.__subclasses__()).union(
         [s for c in cls.__subclasses__() for s in all_subclasses(c)]
     )
+
+
+_have_warned_psutil = False
+
+
+def resize_dask_cache(nbytes: int = None):
+    """Create or resize the dask cache for opportunistic caching.
+
+    The cache object is an instance of a :class`cachey.Cache`, and is made
+    available at :attr:`napari.utils.dask_cache`.
+
+    See `Dask oportunistic caching
+    <https://docs.dask.org/en/latest/caching.html>`_
+
+    Parameters
+    ----------
+    nbytes : int, optional
+        The desired size of the cache, in bytes, by default, napari wil attempt
+        to import ``psutil`` and use 80% of the available memory.  Otherwise it
+        will emit a warning and use 32GB.
+    """
+
+    from dask.cache import Cache
+    from napari import utils
+
+    if not nbytes:
```

Comment:
Maybe we should allow `nbytes=0` to turn off caching, and `nbytes=None` to mean "guess".